internal/app/query/query_service.go

```diff
@@ -325,3 +371,146 @@ func (tc *ticketCache) update() {
 	logger.Debugf("Ticket Cache update: Previous %d, Deleted %d, Fetched %d, Current %d", previousCount, deletedCount, len(toFetch), len(tc.tickets))
 	tc.err = nil
 }
+
+type backfillCache struct {
+	store statestore.Service
+
+	wg sync.WaitGroup
+
+	// backfill cache
+	backfills map[string]*pb.Backfill
+
+	// startRunRequest acts as a semaphore indicator for defining
+	// when the backfillCache is being used (queried) or is it in an idle state.
+	startRunRequest chan struct{}
+
+	// requests is the channel to handle all incoming query requests and
+	// make only one request to the storage service.
+	requests chan *cacheRequest
+
+	// This error is used for the return value when a request fails to reach
+	// the storage. In case of multiple requests are being handled, this error
+	// is returned to every single request.
+	err error
+}
+
+func newBackfillCache(b *appmain.Bindings, cfg config.View) *backfillCache {
+	bc := &backfillCache{
+		store:           statestore.New(cfg),
```

Comment:
Both caches should share the same statestore, move this to be an argument to the newCache, and create it once.